### DISCUSSION TITLE
DG and cohesive zone modelling

### DISCUSSION THREAD

**Discussion by AbbBallout** [idaholab/moose#24047] [Q&A Getting Started] [answered]
I am looking for a FEM library that allows me to do cohesive zone modelling. 
I found the paper "An IIPG-Based Finite Element Framework in MOOSE for Modeling Fiber Reinforced Composite Failure Governed by Extrinsic Cohesive Laws" which makes MOOSE look promising. 

I can't find tutorials in MOOSE for CZM. Are there any examples somewhere else

**Comment by lindsayad** [comment, marked as answer]
There are CZM tests in the `tensor_mechanics` module, see for example
```
lindad@pop-os:~/projects/moose/modules/tensor_mechanics/test/tests(nodal-bc-flex-23940)$ git grep -I --name-only Cohesive
cohesive_zone_model/bilinear_mixed.i
cohesive_zone_model/bilinear_mixed_scale_strength.i
cohesive_zone_model/czm_multiple_action_and_materials.i
cohesive_zone_model/czm_multiple_dimension_base.i
cohesive_zone_model/czm_patch_test_base.i
cohesive_zone_model/czm_traction_separation_base.i
cohesive_zone_model/stretch_rotate_large_deformation.i
```

**Comment by AbbBallout** [comment]
One more thing. 
Are cohesive elements created automatically or should I include those by myself?

**Reply by hugary1995** [reply]
They are created by one of the mesh generator. Please follow the examples @lindsayad provided above.
